google/auth/external_account.py

```diff
@@ -393,12 +394,19 @@ def get_project_id(self, request):
     @_helpers.copy_docstring(credentials.Credentials)
     def refresh(self, request):
         scopes = self._scopes if self._scopes is not None else self._default_scopes
+        auth_request = request
```

Comment:
nit: I don't think it is necessary to create a new variable.

Reply:
If we just set modify the input request, won't that change the behavior of the request instance after the refresh is called with x509? i.e. if someone is re-using the same request for two credential refresh calls, one for X509, and one for file based